src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -695,6 +696,72 @@ private void ScanLdarg (Instruction operation, Stack<StackSlot> currentStack, Me
 			StoreMethodLocalValue (locals, valueToStore.Value, localDef, curBasicBlock);
 		}
 
+		protected abstract ValueNode GetFieldValue (MethodDefinition method, ValueNode instanceValue, FieldDefinition field);
+
+		private void ScanLdsfld (
```

Comment:
Nit: ScanLdsfld/ScanLdfld are pretty much the same. I think we could just add a single `if` to `ScanLdfld` and have it handle both (same for the field stores).